Lib/pydoc.py

```diff
@@ -2544,9 +2544,38 @@ def bltinlink(name):
             'key = %s' % key, '#ffffff', '#ee77aa', '<br>'.join(results))
         return 'Search Results', contents
 
+    def validate_source_path(path):
+        for importer, modname, ispkg in pkgutil.walk_packages():
```

Comment:
``pkgutil.walk_packages()`` can seemingly fail if a module has no ``__version__``. Eg. on my computer I get ``AttributeError`` for some weird packages:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "......\lib\pkgutil.py", line 92, in walk_packages
    __import__(info.name)
  File "......\lib\site-packages\brotli_asgi\__init__.py", line 12, in <module>
    brotli = import_brotli()
  File "......\lib\site-packages\brotli_asgi\_import_resolver.py", line 22, in import_brotli
    spec.loader.exec_module(brotli)
  File "......\lib\site-packages\brotli.py", line 12, in <module>
    __version__ = _brotli.__version__
AttributeError: module 'brotli._brotli' has no attribute '__version__'
```
Because:
```python
>>> import brotli
>>> brotli._brotli
<module 'brotli._brotli' from '......\\lib\\site-packages\\brotli\\_brotli.cp38-win_amd64.pyd'>
>>> brotli._brotli.__version__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'brotli._brotli' has no attribute '__version__'
```